kyaml/filesys/fsondisk_test.go

```diff
@@ -1,137 +1,164 @@
+	require.NoError(t, err)
+
+	cleanedWd, err := filepath.EvalSymlinks(wd)
+	require.NoError(t, err)
+
+	return cleanedWd
+}
+
 func TestCleanedAbs_1(t *testing.T) {
+	req := require.New(t)
 	fSys, _ := makeTestDir(t)
 
 	d, f, err := fSys.CleanedAbs("")
-	if err != nil {
-		t.Fatalf("unexpected err=%v", err)
-	}
-	wd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("unexpected err=%v", err)
-	}
-	if d.String() != wd {
-		t.Fatalf("unexpected d=%s", d)
-	}
-	if f != "" {
-		t.Fatalf("unexpected f=%s", f)
-	}
+	req.NoError(err)
+
+	wd := cleanWd(t)
+	req.Equal(wd, d.String())
+	req.Empty(f)
 }
 
 func TestCleanedAbs_2(t *testing.T) {
+	req := require.New(t)
 	fSys, _ := makeTestDir(t)
 
-	d, f, err := fSys.CleanedAbs("/")
-	if err != nil {
-		t.Fatalf("unexpected err=%v", err)
-	}
-	if d != "/" {
-		t.Fatalf("unexpected d=%s", d)
-	}
-	if f != "" {
-		t.Fatalf("unexpected f=%s", f)
-	}
+	root := getOSRoot(t)
+	d, f, err := fSys.CleanedAbs(root)
+	req.NoError(err)
+	req.Equal(root, d.String())
+	req.Empty(f)
 }
 
 func TestCleanedAbs_3(t *testing.T) {
+	req := require.New(t)
 	fSys, testDir := makeTestDir(t)
 
 	err := fSys.WriteFile(
 		filepath.Join(testDir, "foo"), []byte(`foo`))
-	if err != nil {
-		t.Fatalf("unexpected err=%v", err)
-	}
+	req.NoError(err)
 
 	d, f, err := fSys.CleanedAbs(filepath.Join(testDir, "foo"))
-	if err != nil {
-		t.Fatalf("unexpected err=%v", err)
-	}
-	if d.String() != testDir {
-		t.Fatalf("unexpected d=%s", d)
-	}
-	if f != "foo" {
-		t.Fatalf("unexpected f=%s", f)
-	}
+	req.NoError(err)
+	req.Equal(testDir, d.String())
+	req.Equal("foo", f)
 }
 
 func TestCleanedAbs_4(t *testing.T) {
+	req := require.New(t)
 	fSys, testDir := makeTestDir(t)
 
 	err := fSys.MkdirAll(filepath.Join(testDir, "d1", "d2"))
-	if err != nil {
-		t.Fatalf("unexpected err=%v", err)
-	}
+	req.NoError(err)
+
 	err = fSys.WriteFile(
 		filepath.Join(testDir, "d1", "d2", "bar"),
 		[]byte(`bar`))
-	if err != nil {
-		t.Fatalf("unexpected err=%v", err)
-	}
+	req.NoError(err)
 
 	d, f, err := fSys.CleanedAbs(
 		filepath.Join(testDir, "d1", "d2"))
-	if err != nil {
-		t.Fatalf("unexpected err=%v", err)
-	}
-	if d.String() != filepath.Join(testDir, "d1", "d2") {
-		t.Fatalf("unexpected d=%s", d)
-	}
-	if f != "" {
-		t.Fatalf("unexpected f=%s", f)
-	}
+	req.NoError(err)
+	req.Equal(filepath.Join(testDir, "d1", "d2"), d.String())
+	req.Empty(f)
 
 	d, f, err = fSys.CleanedAbs(
 		filepath.Join(testDir, "d1", "d2", "bar"))
-	if err != nil {
-		t.Fatalf("unexpected err=%v", err)
-	}
-	if d.String() != filepath.Join(testDir, "d1", "d2") {
-		t.Fatalf("unexpected d=%s", d)
+	req.NoError(err)
+	req.Equal(filepath.Join(testDir, "d1", "d2"), d.String())
+	req.Equal("bar", f)
+}
+
+func TestConfirmDirDisk(t *testing.T) {
+	req := require.New(t)
+	fSys, testDir := makeTestDir(t)
+	wd := cleanWd(t)
+
+	relDir := "actual_foo_431432"
+	err := fSys.Mkdir(relDir)
+	req.NoError(err)
+	req.Truef(fSys.Exists(relDir), existMsg, relDir)
+	t.Cleanup(func() {
```

Comment:
move this up to L114. With the order you have now, if L115 fails, the cleanup function is never called